AppCenter/AppCenter/Internals/Model/Util/MSAppCenterUserDefaults.m

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "MSAppCenterUserDefaults.h"
+#import "MSAppCenterInternal.h"
+#import "MSLogger.h"
+
+static NSString *const kMSAppCenterUserDefaultsMigratedKey = @"310UserDefaultsMigratedKey";
+
+static MSAppCenterUserDefaults *sharedInstance = nil;
+static dispatch_once_t onceToken;
+
+// MSAppCenterUserDefaults keys to be migrated.
+static NSMutableDictionary<NSString *, NSString *> *keysToMigrate;
+
+@implementation MSAppCenterUserDefaults
+
++ (void)load {
+  keysToMigrate = [NSMutableDictionary new];
+}
+
++ (void)addKeysToMigrate:(NSDictionary<NSString *, NSString *> *)keys {
+  [keysToMigrate addEntriesFromDictionary:keys];
+}
+
++ (instancetype)shared {
+  dispatch_once(&onceToken, ^{
+    sharedInstance = [[MSAppCenterUserDefaults alloc] init];
+    NSDictionary *changedKeys = @{
+      @"MSChannelStartTimer" : @"MSAppCenterChannelStartTimer",         // MSChannelUnitDefault
```

Comment:
This is tricky because it's per channels so the key is dynamic[ look here](https://github.com/microsoft/appcenter-sdk-apple/blob/8e1835b06ae67054a091bb826d7faaed1ad29628/AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m#L409). I'd say you can scan the user defaults, list keys prefixed with `MSChannelStartTimer` and migrate.